js_src/_dart_helpers.js

```diff
@@ -6,6 +6,11 @@ var _reactDartSymbolPrefix = 'react-dart.';
 /// A global symbol to identify javascript objects owned by react-dart context,
 /// in order to jsify and unjsify context objects correctly.
 var _reactDartContextSymbol = Symbol(_reactDartSymbolPrefix+'context');
+
```

Comment:
I added a note in `componentDidCatch` about adding a comment to explain the purpose. This kind of goes in hand in hand with that - I think a comment here explaining `_rethrowError` could be useful.